src/App.js

```diff
@@ -1,15 +1,50 @@
 import MainPage from './pages/MainPage'
 import { Box } from '@chakra-ui/react'
-import React from 'react'
+import React, { useContext, useState } from 'react'
 import cardData from './data/CardData.json'
 
+const cardDataContext = React.createContext()
+
+const CardDataProvider = ({ children }) => {
+    const [dataDisplay] = useState(cardData)
+    const [checkedCard, setCheckedCard] = useState({})
+    const [numberOfCards, setNumberOfCards] = useState(dataDisplay.length)
```

Comment:
Why do you need state for `numberOfCards`? I expect to see one method like `getCardsCount`